experimental/oak_baremetal_runtime/tests/integration_test.rs

```diff
@@ -0,0 +1,163 @@
+        let body = builder.create_vector::<u8>(&client_hello);
+        let flatbuffer = schema::UserRequest::create(
+            &mut builder,
+            &schema::UserRequestArgs {
+                session_id: Some(session_id),
+                body: Some(body),
+            },
+        );
+        builder
+            .finish(flatbuffer)
+            .map_err(|err| {
+                oak_idl::Status::new_with_message(oak_idl::StatusCode::Internal, err.to_string())
+            })
+            .unwrap()
+    };
+    let result = client.handle_user_request(owned_request_flatbuffer.into_vec());
+
+    assert_eq!(
+        result.unwrap_err(),
+        oak_idl::Status::new(oak_idl::StatusCode::FailedPrecondition)
+    );
+}
+
+#[test]
+fn it_should_handle_user_requests_after_initialization() {
+    let attestation_behavior =
+        AttestationBehavior::create(PlaceholderAmdAttestationGenerator, EmptyAttestationVerifier);
+    let runtime = oak_baremetal_runtime::RuntimeImplementation::new(attestation_behavior.clone());
+    let mut client = schema::TrustedRuntimeClient::new(TrustedRuntime::serve(runtime));
+
+    let owned_initialization_flatbuffer = {
+        let mut builder = oak_idl::utils::OwnedFlatbufferBuilder::default();
+        let wasm_module = builder.create_vector::<u8>(ECHO_WASM_BYTES);
+        let initialization_flatbuffer = schema::Initialization::create(
+            &mut builder,
+            &schema::InitializationArgs {
+                wasm_module: Some(wasm_module),
+                constant_response_size: MOCK_CONSTANT_RESPONSE_SIZE,
+            },
+        );
+
+        builder
+            .finish(initialization_flatbuffer)
+            .expect("errored when creating initialization message")
+    };
+
+    client
+        .initialize(owned_initialization_flatbuffer.into_vec())
+        .unwrap();
+
+    let mut handshaker =
+        ClientHandshaker::new(attestation_behavior).expect("could not create client handshaker");
+    let client_hello = handshaker
+        .create_client_hello()
+        .expect("Couldn't create client hello message");
+
+    let owned_request_flatbuffer = {
+        let mut builder = oak_idl::utils::OwnedFlatbufferBuilder::default();
+        let session_id = &schema::SessionId::new(MOCK_SESSION_ID);
+        let body = builder.create_vector::<u8>(&client_hello);
+        let flatbuffer = schema::UserRequest::create(
+            &mut builder,
+            &schema::UserRequestArgs {
+                session_id: Some(session_id),
+                body: Some(body),
+            },
+        );
+        builder
+            .finish(flatbuffer)
+            .map_err(|err| {
+                oak_idl::Status::new_with_message(oak_idl::StatusCode::Internal, err.to_string())
+            })
+            .unwrap()
+    };
+    let result = client.handle_user_request(owned_request_flatbuffer.into_vec());
+
+    assert!(result.is_ok());
```

Comment:
You can try `assert_matches!(result, Ok(_))`, then tests will show the exact error from the result if it is one